site/src/pages/RoadmapPage/CourseYear.tsx

```diff
@@ -0,0 +1,152 @@
+import React, { FC, useState } from 'react';
+import { Button, Form, Modal } from 'react-bootstrap';
+import { PlannerYearData } from '../../types/types';
+
+interface YearPopupQuarter {
+  id: string;
+  name: string;
+  checked?: boolean;
+}
+
+interface CourseYearModalProps {
+  placeholderName: string;
+  placeholderYear: number;
+  show: boolean;
+  setShow: React.Dispatch<React.SetStateAction<boolean>>;
+  type: 'add' | 'edit';
+  saveHandler: (x: PlannerYearData) => void;
+  currentQuarters: string[];
+}
+
+const CourseYearModal: FC<CourseYearModalProps> = (props) => {
+  const { placeholderName, placeholderYear, show, setShow, type, saveHandler, currentQuarters } = props;
+  const [validated, setValidated] = useState(false);
+
+  const [name, setName] = useState(placeholderName);
+  const [year, setYear] = useState(placeholderYear);
+
+  const quarterValues: (selectedNames: string[]) => YearPopupQuarter[] = (data: string[]) => {
+    const base: YearPopupQuarter[] = [
+      { id: 'fall', name: 'Fall' },
+      { id: 'winter', name: 'Winter' },
+      { id: 'spring', name: 'Spring' },
+      { id: 'summer 1', name: 'Summer 1' },
+      { id: 'summer 2', name: 'Summer 2' },
+      { id: 'summer 10 Week', name: 'Summer 10 Week' },
+    ];
+    data.forEach((name) => {
+      const match = base.find((q) => q.id === name);
+      if (match) match.checked = true;
+    });
+    return base;
+  };
+
+  const [quarters, setQuarters] = useState<YearPopupQuarter[]>(quarterValues(currentQuarters));
+  const quarterCheckboxes = quarters.map((q, i) => {
+    const handleClick = (i: number) => {
+      quarters[i].checked = !quarters[i].checked;
+      setQuarters(quarters.slice());
+    };
+    return (
+      <Form.Check
+        key={q.id}
+        type="checkbox"
+        id={'quarter-checkbox-' + q.id}
+        label={q.name}
+        value={q.id}
+        checked={q.checked ?? false}
+        onChange={() => handleClick(i)}
+      />
+    );
+  });
+
+  const title = type === 'add' ? 'Add Year' : `Editing "${placeholderName}"`;
```

Comment:
This runs into the same issue from implementing #368 where for schedules before custom year names were added, their name is undefined. We had it fall back to using the `Year {index + 1}` format in the implementation but given that we are running into this problem again and possibly will in the future, I've drafted up a script that should add default names to any planner with years that are missing names. On first test, it looks good but I might want to find a way to back up the database before proceeding to modify prod. I'll keep you posted.

```ts
import { MongoClient, ObjectId } from 'mongodb';
import dotenv from 'dotenv';

dotenv.config();

const client = new MongoClient(process.env.MONGO_URL);
const DB_NAME = 'peterPortalDevDB';
const db = client.db(DB_NAME);
const roadmaps = db.collection('roadmaps');
const documents = roadmaps.find();

interface Year {
  startYear: number;
  name?: string;
  quarters: object[];
}

interface Roadmap {
  planner: Year[];
  transfers: object[];
}

interface RoadmapDoc {
  _id: ObjectId;
  roadmap: Roadmap;
}

let document = await documents.next() as RoadmapDoc;

while (document != null) {
  let planner = document.roadmap.planner;
  for (let i = 0; i < planner.length; i++) {
    if (planner[i].name == null) {
      planner[i].name = 'Year ' + (i + 1);
    }
  }
  roadmaps.updateOne({ _id: document._id }, {
    $set: {
      roadmap: document.roadmap
    }
  })
  document = await documents.next() as RoadmapDoc;
}

console.log('done')
```

Reply:
I just copied what was used for the popup before, but if I understand correctly, you're saying that `placeholderName` may be undefined for people who created roadmaps a long time ago and that we should have a fallback as "Year {index + 1}" right?

Reply:
I don't have access to `yearIndex` from within the modal so I am putting it in the call to the Modal in `Year.tsx`

Reply:
> I just copied what was used for the popup before, but if I understand correctly, you're saying that `placeholderName` may be undefined for people who created roadmaps a long time ago and that we should have a fallback as "Year {index + 1}" right?

Yes, any roadmaps made before #368 may not have year names resulting in `placeholderName` being undefined.